walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,55 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
Where is the validation for the `noOptionsForTooLongOfARequest` test?
I think the test passes because of the empty events being passed?

Seems also the same for the `optionsForNoAttendees` test.

Reply:
Is it necessary to have an extra validation for these two edge cases, when my original algorithm already handles it the way the tests want it to be?

Reply:
This is a fair argument assuming that the test cases perfectly cover the possible scenarios.

If you look at those test cases, the `query` method is passed in `NO_EVENTS`.
What if there are some events scheduled and passed to the `query` method?
Does your logic cover that?

Reply:
Added extra tests to show that these cases are covered, even if there are events passed in to queries.

Reply:
Thanks for adding the extra test cases.

Although the logic handles it well, I think it can be improved further by doing it as separate validation logic at the beginning.
The current approach needs to iterate all the events unnecessarily for a meeting request that has no attendee or the duration is too long.